mutation_reader.hh

```diff
@@ -273,22 +293,24 @@ public:
         schema_ptr s,
```

Comment:
Maybe we should also remove all the `mutation_source()` constructors that take less than the entire parameter pack. Not entirely convinced though.